### ISSUE TITLE
md-dialog issues with scrim and escape

### ISSUE THREAD

**Issue by MarkStega** [material-components/material-web#5258]
### What is affected?

Component

### Description

1) It is desirable to be able to disable a click on the scrim or use of the escape key to close certain dialogs. In the m2.material.io implementation properties of scrimClickAction and  escapeKeyAction were available. The same should be supported in some fashion with md-dialog.

2) If you show a dialog and close it with either the scrim or escape then the return value is an empty string. If you then show the same dialog and press a button associated with the form the returnValue is the value specified by the value of the button. If you then show that same dialog and close with either a scrim click or escape the returnValue has the value from the second dialog show.

It feels like it shouldn't be the responsibility of the invoking code to clear the returnValue when a dialog is first shown.

### Reproduction

This behavior is demonstrated in the material-blazor/material.blazor repository. You need to checkout the vNext branch and use your favorite method (command line, Visual Code, Visual Studio, etc.) to run the material.blazor.website.server.md3 project and then pick the 'dialog' choice from the dropdown menu.

### Workaround

None known

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

Failing in 1.0.2-nightly.a0baa4d.0

### Browser/OS/Node environment

Browser: Chrome Version 119.0.6045.200 (Official Build) (64-bit)
OS: Windows 10 Pro
Node: 20.1.0

**Comment by markope** [comment]
When pressing escape key first then you can do `event.preventDefault()` and the dialog stays open. When you hit escape another time however, then something else seems to have focus and the dialog closes. To prevent this I do: 

```javascript
dialog.addEventListener('cancel', event => {
  event.preventDefault(); // Stop scrim click from closing dialog
});
dialog.addEventListener('keydown', event => {
  if (event.key === 'Escape') event.preventDefault(); // Stop cancelation by escape key
});
```

**Comment by asyncLiz** [comment]
Pressing escape twice without changing focus after a dialog has been prevented from closing should trigger browsers' dialog abuse heuristics, which will forcefully close the dialog and may not be canceled.

This heuristic is designed to prevent `<dialog>` from remaining open after multiple repeated Escape/Back gestures, and is related to the new CloseWatcher API. Here's a link with some more history on this topic: https://issues.chromium.org/issues/41484805

I wouldn't suggest the keydown preventDefault workaround. It won't work with other dialog cancelation gestures, such as pressing the back button or making a back swipe gesture on mobile. It could also stop working in the future if its patched, or it could be inconsistent across browsers.

Instead, I'd recommend handling when a dialog is forcibly closed in your event listener. You should be able to check `event.cancelable` to know if its safe to call preventDefault().

```js
dialog.addEventListener('cancel', event => {
  if (event.cancelable) {
    event.preventDefault(); // Stop scrim click from closing dialog
  } else {
    // The browser is forcing the dialog to close.
  }
});
```

tl;dr; this happens because browsers give users a way to force close dialogs
